x-pack/plugins/security_solution/public/management/components/console/components/command_input/command_input.test.tsx

```diff
@@ -16,144 +16,274 @@ describe('When entering data into the Console input', () => {
-      renderWithInputHistory('one');
+      await renderWithInputHistory('one');
 
-      waitFor(() => {
-        expect(renderResult.getByTestId('test-inputPlaceholder').textContent).toEqual(
-          'cmd1 --help'
-        );
+      await waitFor(() => {
+        expect(getInputPlaceholderText()).toEqual('cmd1 --help');
       });
 
       userEvent.keyboard('{Escape}');
 
-      waitFor(() => {
-        expect(renderResult.getByTestId('test-inputPlaceholder').textContent).toEqual('one');
+      await waitFor(() => {
+        expect(getUserInputText()).toEqual('one');
       });
     });
 
     it('should add history item to the input area when selected and clear placeholder', async () => {
-      renderWithInputHistory('one');
+      await renderWithInputHistory('one');
 
-      waitFor(() => {
-        expect(renderResult.getByTestId('test-inputPlaceholder').textContent).toEqual(
-          'cmd1 --help'
-        );
+      await waitFor(() => {
+        expect(getInputPlaceholderText()).toEqual('cmd1 --help');
       });
 
       userEvent.keyboard('{Enter}');
 
-      waitFor(() => {
-        expect(renderResult.getByTestId('test-cmdInput-userTextInput').textContent).toEqual(
-          'cmd1 --help'
-        );
+      await waitFor(() => {
+        expect(getUserInputText()).toEqual('cmd1 --help');
       });
     });
   });
+
+  describe('and keyboard special keys are pressed', () => {
+    const getRightOfCursorText = () => {
+      return renderResult.getByTestId('test-cmdInput-rightOfCursor').textContent;
+    };
+
+    const typeKeyboardKey = (key: string) => {
+      enterCommand(key, { inputOnly: true, useKeyboard: true });
+    };
+
+    beforeEach(() => {
+      render();
+      enterCommand('isolate', { inputOnly: true });
+    });
+
+    it('should backspace and delete last character', () => {
+      typeKeyboardKey('{backspace}');
+      expect(getUserInputText()).toEqual('isolat');
+      expect(getRightOfCursorText()).toEqual('');
+    });
+
+    it('should move cursor to the left', () => {
+      typeKeyboardKey('{ArrowLeft}');
+      typeKeyboardKey('{ArrowLeft}');
+      expect(getUserInputText()).toEqual('isola');
+      expect(getRightOfCursorText()).toEqual('te');
+    });
+
+    it('should move cursor to the right', () => {
+      typeKeyboardKey('{ArrowLeft}');
+      typeKeyboardKey('{ArrowLeft}');
+      expect(getUserInputText()).toEqual('isola');
+      expect(getRightOfCursorText()).toEqual('te');
+
+      typeKeyboardKey('{ArrowRight}');
+      expect(getUserInputText()).toEqual('isolat');
+      expect(getRightOfCursorText()).toEqual('e');
+    });
+
+    it('should move cursor to the beginning', () => {
+      typeKeyboardKey('{Home}');
+      expect(getUserInputText()).toEqual('');
+      expect(getRightOfCursorText()).toEqual('isolate');
+    });
+
+    it('should should move cursor to the end', () => {
```

Comment:
```suggestion
    it('should move cursor to the end', () => {
```

Reply:
crap. I'll correct it in a subsequent PR